xarray_subset_grid/grids/ugrid.py

```diff
@@ -113,31 +112,54 @@ def subset_polygon(
         x_var, y_var = mesh.node_coordinates.split(" ")
         x, y = ds[x_var], ds[y_var]
 
+        # NOTE: When the first dimension is "nele", the face_node_connectivity
```

Comment:
According to the UGRID spec:
"""
The connectivity array will thus be a matrix of size nFaces x 3. 
"""
so the "nele" dimension should always be first in a compliant file.

If we are finding files that. need to be transposed (ADCIRC?) -- what to do?

I think that should be handled by a pre-processor -- the make compliant function:

`assign_ugrid_topology`

If we are consistent about that, then we'll save a lot if blocks in the core code, like here.

Reply:
I do not think the preprocessor should mess with the dataset itself personally, but I can be convinced maybe.

Yes the ADCIRC files have reversed dimensions on the face node connectivity, which could happen with any dataset really.

Reply:
You are probably right -- as much as I want to be pedantic about the standard, we shouldn't actually change anything (as opposed to adding metadata) in the process unless the user asks for it ...